src/nnet3/nnet-diagnostics.h

```diff
@@ -168,14 +186,27 @@ class NnetComputeProb {
    @param [out] tot_weight  The sum of the values in the supervision matrix
    @param [out] tot_accuracy  The total accuracy, equal to the sum over all row
                      indexes r such that the maximum column index of row r of
-                     supervision and nnet_output is the same, of the sum of the
-                     r'th row of supervision (i.e. the row's weight).
+                     supervision and nnet_output is the same, of the sum of 
+                     the r'th row of supervision (i.e. the row's weight).
+   @param [out] *tot_weight_vec  We write to this location
```

Comment:
I don't recognize the star ('*') as something I've ever seen before in these documentation
strings;  please remove even if it's technically allowed.  Explain that these are written to if non-NULL.